core/control-plane/control-plane-api/src/main/java/org/eclipse/edc/connector/api/transferprocess/TransferProcessControlApiController.java

```diff
@@ -34,25 +35,33 @@
 public class TransferProcessControlApiController implements TransferProcessControlApi {
 
     public static final String PATH = "/transferprocess";
-    private final TransferProcessManager transferProcessManager;
+    private final TransferProcessService transferProcessService;
 
 
-    public TransferProcessControlApiController(TransferProcessManager transferProcessManager) {
-        this.transferProcessManager = transferProcessManager;
+    public TransferProcessControlApiController(TransferProcessService transferProcessService) {
+        this.transferProcessService = transferProcessService;
     }
 
 
     @POST
     @Path("/{processId}/complete")
     @Override
     public void complete(@PathParam("processId") String processId) {
-        transferProcessManager.enqueueCommand(new CompleteTransferCommand(processId));
+        var result = transferProcessService.complete(processId);
+
+        if (result.failed()) {
+            throw mapToException(result, TransferProcess.class, processId);
+        }
     }
 
     @POST
     @Path("/{processId}/fail")
     @Override
     public void fail(@PathParam("processId") String processId, @NotNull @Valid TransferProcessFailStateDto request) {
-        transferProcessManager.enqueueCommand(new FailTransferCommand(processId, request.getErrorMessage()));
+        var result = transferProcessService.fail(processId, request.getErrorMessage());
+
+        if (result.failed()) {
+            throw mapToException(result, TransferProcess.class, processId);
```

Comment:
a proposal, could the `mapToException` accepts only the `failure` object without all the wrapping `result`?

Reply:
Yes it would be more consistent actually, as of now if we call `mapToException` with success it will throw an `NPE` :D